drivers/input/input_cst816s.c

```diff
@@ -292,14 +327,104 @@ static int cst816s_init(const struct device *dev)
 	return ret;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int cst816s_apply_profile(const struct cst816s_config *cfg,
+				 const struct cst816s_lp_profile *p)
+{
+	int ret;
+
+	ret = i2c_reg_write_byte_dt(&cfg->i2c, CST816S_REG_AUTOSLEEP_TIME, p->autosleep_time_s);
+	if (ret) {
+		LOG_WRN("Write autosleep_time 0x%02x failed (%d)", p->autosleep_time_s, ret);
+		return ret;
+	}
+
+	ret = i2c_reg_write_byte_dt(&cfg->i2c, CST816S_REG_LP_AUTO_WAKEUP_TIME,
+				    p->lp_auto_wake_time_min);
+	if (ret) {
+		LOG_WRN("Write lp_auto_wake_time_min 0x%02x failed (%d)", p->lp_auto_wake_time_min,
+			ret);
+		return ret;
+	}
+
+	ret = i2c_reg_write_byte_dt(&cfg->i2c, CST816S_REG_LP_SCAN_TH, p->lp_scan_th);
+	if (ret) {
+		LOG_WRN("Write lp_scan_th 0x%02x failed (%d)", p->lp_scan_th, ret);
+		return ret;
+	}
+
+	ret = i2c_reg_write_byte_dt(&cfg->i2c, CST816S_REG_LP_SCAN_WIN, p->lp_scan_win);
+	if (ret) {
+		LOG_WRN("Write lp_scan_win 0x%02x failed (%d)", p->lp_scan_win, ret);
+		return ret;
+	}
+
+	ret = i2c_reg_write_byte_dt(&cfg->i2c, CST816S_REG_LP_SCAN_FREQ, p->lp_scan_freq);
+	if (ret) {
+		LOG_WRN("Write lp_scan_freq 0x%02x failed (%d)", p->lp_scan_freq, ret);
+		return ret;
+	}
+
+	ret = i2c_reg_write_byte_dt(&cfg->i2c, CST816S_REG_LP_SCAN_I_DAC, p->lp_scan_i_dac);
+	if (ret) {
+		LOG_WRN("Write lp_scan_i_dac 0x%02x failed (%d)", p->lp_scan_i_dac, ret);
+		return ret;
+	}
```

Comment:
#define CST816S_REG_LP_AUTO_WAKEUP_TIME 0xF4
#define CST816S_REG_LP_SCAN_TH          0xF5
#define CST816S_REG_LP_SCAN_WIN         0xF6
#define CST816S_REG_LP_SCAN_FREQ        0xF7
#define CST816S_REG_LP_SCAN_I_DAC       0xF8
#define CST816S_REG_AUTOSLEEP_TIME      0xF9

Since they are continous can you maybe reorder the struct fields and write the entire struct at once?